tests/by-util/test_timeout.rs

```diff
@@ -143,18 +143,29 @@ fn test_kill_after_long() {
         .no_stderr();
 }
 
+/// Test that a command that is executed in a sub-shell is killed/terminated.
 #[test]
-fn test_kill_subprocess() {
+fn test_kill_subshell() {
     new_ucmd!()
         .args(&[
-            // Make sure the CI can spawn the subprocess.
-            "10",
+            "1",
```

Comment:
I am not sure what the convetion here is. How large should the timeout be? In my eyes: it can be as short as possible, but still has to wait for the `trap ...` command to finish executing, and then interrupt the `sleep` execution afterwards. I think the previous value of 10 was excessive though.

Reply:
Right, there's no real guideline, but I guess it should be as short as possible while also keeping in mind that CI and code coverage make the tests much slower.

Reply:
Can you give an estimate of how much slower/can I test that anywhere. Naively, I would think 1 second is long enough for the sub-shell to startup and execute the `trap` command. I see a lot of other tests using `1` as the timeout value, but there the executed command doesn't rely on timing (it doesn't rely matter if the subprocess starts up properly, it should be killed either way), but for this test case, if the sub-shell is killed before the trap command is executed, it won't return the `xyz` to stdout, and therefore the test would fail, and become flaky. 
So can is there a way to test-run the CI? I could increase the `1` to `3` maybe?

Reply:
Ok, nevermind, I just had a look at the automatically executed tests, and looks like the test really is flaky for the exact reason I described, the sub-shell must be killed before it can execute the `trap` command. I'll increase the timeout then to 5 and try again.

Reply:
Yeah it's a trial and error process. All I can say is that flaky tests are much more annoying than waiting a second 😄